x-pack/legacy/plugins/siem/public/containers/case/use_delete_cases.tsx

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useCallback, useReducer } from 'react';
+import { errorToToaster } from '../../components/ml/api/error_to_toaster';
+import * as i18n from './translations';
+import { useStateToaster } from '../../components/toasters';
+import { deleteCases } from './api';
+
+interface DeleteState {
+  isDisplayConfirmDeleteModal: boolean;
+  isDeleted: boolean;
+  isLoading: boolean;
+  isError: boolean;
+}
+type Action =
+  | { type: 'DISPLAY_MODAL'; payload: boolean }
+  | { type: 'FETCH_INIT' }
+  | { type: 'FETCH_SUCCESS'; payload: boolean }
+  | { type: 'FETCH_FAILURE' }
+  | { type: 'RESET_IS_DELETED' };
+
+const dataFetchReducer = (state: DeleteState, action: Action): DeleteState => {
+  switch (action.type) {
+    case 'DISPLAY_MODAL':
+      return {
+        ...state,
+        isDisplayConfirmDeleteModal: action.payload,
+      };
+    case 'FETCH_INIT':
+      return {
+        ...state,
+        isLoading: true,
+        isError: false,
+      };
+    case 'FETCH_SUCCESS':
+      return {
+        ...state,
+        isLoading: false,
+        isError: false,
+        isDeleted: action.payload,
+      };
+    case 'FETCH_FAILURE':
+      return {
+        ...state,
+        isLoading: false,
+        isError: true,
+      };
+    case 'RESET_IS_DELETED':
+      return {
+        ...state,
+        isDeleted: false,
+      };
+    default:
+      throw new Error();
+  }
+};
+interface UseDeleteCase extends DeleteState {
+  dispatchResetIsDeleted: () => void;
+  handleOnDeleteConfirm: (caseIds: string[]) => void;
+  handleToggleModal: () => void;
+}
+
+export const useDeleteCases = (): UseDeleteCase => {
```

Comment:
What do you think about creating an AbortController and pass it to the API calls?

Reply:
show me?

Reply:
```
  const dispatchDeleteCases = useCallback(async (caseIds: string[]) => {
    let cancel = false;
    const abortCtrl = new AbortController();
    try {
      dispatch({ type: 'FETCH_INIT' });
      await deleteCases({ caseIds, signal: abortCtrl.signal });
      if (!cancel) {
        dispatch({ type: 'FETCH_SUCCESS', payload: true });
      }
    } catch (error) {
      if (!cancel) {
        errorToToaster({
          title: i18n.ERROR_TITLE,
          error: error.body && error.body.message ? new Error(error.body.message) : error,
          dispatchToaster,
        });
        dispatch({ type: 'FETCH_FAILURE' });
      }
    }
    return () => {
      cancel = true;
      abortCtrl.abort();
    };
  }, []);
```

Reply:
@XavierM thoughts?

Reply:
That's correct, I notice that too but I was thinking that we can change it to a specific PR later on and create an issue to track it. therefore we can move forward to this PR and staying true to just do deletions and showing comments. I am saying that because that's not the only places we will have to change it. does that make sense?